### PR TITLE
giada: unstable-2021-09-24 -> 0.24.0

### PR DESCRIPTION
**Pull request by wkral**
###### Description of changes
Update to lastest release and fix the build.

https://github.com/monocasual/giada/releases/tag/v0.24.0

ZHF: #230712

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>giada</li>
  </ul>
</details>

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/giada/default.nix [hunk 1]
```diff
@@ -1,32 +1,36 @@
...
 , libogg
 , libvorbis
 , libopus
+, nlohmann_json
 }:
 
 stdenv.mkDerivation rec {
   pname = "giada";
-  version = "unstable-2021-09-24";
+  version = "v0.24.0";
```
Comment (wegank) [inline pkgs/applications/audio/giada/default.nix:27]:
```suggestion
  version = "0.24.0";
```

File: pkgs/applications/audio/giada/default.nix [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
 
   src = fetchFromGitHub {
     owner = "monocasual";
     repo = pname;
-    # Using master with https://github.com/monocasual/giada/pull/509 till a new release is done.
-    rev = "f117a8b8eef08d904ef1ab22c45f0e1fad6b8a56";
-    sha256 = "01hb981lrsyk870zs8xph5fm0z7bbffpkxgw04hq487r804mkx9j";
+    rev = version;
```
Comment (wegank) [inline pkgs/applications/audio/giada/default.nix:32]:
```suggestion
    rev = "v${version}";
```

File: pkgs/applications/audio/giada/default.nix [hunk 3]
```diff
@@ -58,19 +65,16 @@ stdenv.mkDerivation rec {
...
-      src/core/kernelMidi.cpp
-      src/gui/elems/config/tabMidi.cpp
-      src/utils/ver.cpp
-    )
-    for f in "''${fixup_list[@]}"; do
-      substituteInPlace "$f" \
-        --replace "<RtMidi.h>" "<${rtmidi.src}/RtMidi.h>"
-    done
-  '';
+  patches = [
```
Comment (wegank) [inline pkgs/applications/audio/giada/default.nix:71]: Probably should be placed under `src`.

File: pkgs/applications/audio/giada/default.nix [hunk 4]
```diff
@@ -58,19 +65,16 @@ stdenv.mkDerivation rec {
...
+    # Included in next release https://github.com/monocasual/giada/pull/635
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/giada/default.nix:72]:
```suggestion
    # fix fmt type error https://github.com/monocasual/giada/pull/635
```
if I am guessing that correct

Comment (wkral) [reply pkgs/applications/audio/giada/default.nix]: Wanted to make it clear to remove it on the next update, but added another line along with your suggestion to make it more explicit.

**Comment by wkral** [comment]
Implemented your suggestions, @wegank @SuperSandro2000 

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>giada</li>
  </ul>
</details>

File: pkgs/applications/audio/giada/default.nix [hunk 5]
```diff
@@ -1,35 +1,48 @@
...
+    rev = "v${version}";
+    sha256 = "sha256-pKzc+RRW3o5vYaiGqW9/VjYZZJvr6cg1kdjP9qRkHwM=";
     fetchSubmodules = true;
   };
 
+  patches = [
+    # Remove when updating to the next release, this PR is already merged
+    # Fix fmt type error: https://github.com/monocasual/giada/pull/635
+    (fetchpatch {
+      url = "https://github.com/monocasual/giada/commit/032af4334f6d2bb7e77a49e7aef5b4c4d696df9a.patch";
```
Comment (risicle) [inline pkgs/applications/audio/giada/default.nix:41]:
```suggestion
      name = "fix-fmt-type-error.patch";
      url = "https://github.com/monocasual/giada/commit/032af4334f6d2bb7e77a49e7aef5b4c4d696df9a.patch";
```

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, nixos x86_64 & aarch64-linux

File: pkgs/applications/audio/giada/default.nix (hunk 5 above)
Comment (wkral) [reply pkgs/applications/audio/giada/default.nix:42]: Fixed.

**Pull request merged**